### ISSUE TITLE
MonoInterpreter doesn't correctly handle overshifting of a vector

### ISSUE THREAD

**Issue by tannergooding** [dotnet/runtime#89938]
As found in https://github.com/dotnet/runtime/pull/89339, monointerpreter does not currently correctly mask the `shiftAmount` and produces an incorrect result when the shift amount is larger than the number of bits available in the underlying `T`.

The simplest repro is:
```csharp
Vector128<byte> vector = Vector128.Create(1);
Console.WriteLine(vector << 8);
```

Where the result should be `1, 1, ..., 1`.

 
The same should be true for `Vector128<short>` using `16`, `Vector128<int>` using `32`, and `Vector128<long>` using `64`.

The `shiftAmount` in all cases should be functionally `vector << (shiftAmount & ((sizeof(T) * 8) - 1))`

**Comment by tannergooding** [comment]
Potentially the testcase wasn't complicated enough. The full repro can be seen by removing the `[ActiveIssue]` here and running the tests: https://github.com/dotnet/runtime/blob/main/src/tests/JIT/HardwareIntrinsics/General/Shared/VectorImmBinaryOperatorTest.template#L22

**Comment by SamMonoRT** [comment]
Possibly will be fixed via https://github.com/dotnet/runtime/pull/90666

**Comment by SamMonoRT** [comment]
Fix has been merged to main and also backported to release\8.0 - Closing the issue.
